# hypothesis testing

**Q1.Suppose a child psychologist claims that the average time working mothers spend talking to their children is at least 11 minutes per day. You conduct a random sample of 1000 working mothers and find they spend an average of 11.5 minutes per day talking with their children. Assume prior research suggests the population standard deviation is 2.3 minutes.Conduct a test with a level of significance of alpha = 0.05.**

## State the hypothesis

**null hypothesis**

Ho = μ ≤ 11 minutes



**alternate hypothesis**

Ha = μ > 11 minutes



##  Set the significance level

α = 0.05 minutes



## Calculate the test statistic: We'll use the z-test statistic formula


x̄ = 11.5 minutes (sample mean)
μ = 11 minutes (population mean under the null hypothesis)
σ = 2.3 minutes (population standard deviation)
n = 1000 (sample size)

Z=(x̄ - μ)/(σ/√n)



In [3]:
import math

In [1]:
alpha = 0.05
std = 2.3
n =1000
n_mean = 11.5
populataion_mean =11

In [4]:
#Hypothesis testing

z_test = (n_mean - populataion_mean) / (std/math.sqrt(n))

In [2]:
Z=(11.5-11)/(2.3)*(math.sqrt(1000))
Z

6.874516652539955

In [5]:
import scipy.stats as st


alpha = 0.05


z_critical = st.norm.ppf(1 - alpha)

print("Critical value:", z_critical)

Critical value: 1.6448536269514722


critical value for a z-score corresponding to an alpha of 0.05 from z-distribution table is 1.645.



SINCE OUR CALCULATED Z-VALUE 6.8745 WHICH GREATER THAN THE CRITICAL VALUE 1.645 WE REJECT THE NULL HYPOTHESIS

**Q2. A coffee shop claims that their average wait time for customers is less than 5 minutes. To test this claim, a sample of 40 customers is taken, and their wait times are recorded. The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes. Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim.**

**null hypothesis**

Ho = μ > 5 minutes



**alternate hypothesis**

 Ha = μ <5 minutes



x̄ = 4.6 minutes (sample mean)

μ = 5 minutes (population mean)

σ = 0.8 minutes (sample standard deviation)

n = 40 (sample size)

α = 0.05 (significance level)

Degree of freedom(n-1) = 39

In [6]:
population_tmean = 5
sample_tmean =4.6
sample_size = 40
std_t = 0.8
alpha = 0.05

The t-test works well when we don't know the population standard deviation, especially for smaller groups, usually less than 30. But it's still good to use even for bigger groups, over 30. This is because of the Central Limit Theorem, which says that as our group size gets bigger, our average values act more like they're from a normal distribution. So, no matter if our group is small or big, the t-test is a reliable way to check things.

In [7]:
t_test =  (sample_tmean - population_tmean) / (0.8/math.sqrt(sample_size))
t_test

-3.162277660168382

In [ ]:
from scipy.stats import t

# Degrees of freedom
df = 40 - 1  # Sample size minus 1


alpha = 0.05

# Find critical value
t_critical = t.ppf(1 - alpha, df)

print("Critical value:", t_critical)

t = (x̄ - μ) / (σ / √n)



Since our calculated t-value -3.162 which is less than the critical value -1.645, we reject the null hypothesis.